pkg/crd/markers/crd.go

```diff
@@ -57,6 +57,9 @@ var CRDMarkers = []*definitionWithHelp{
 
 	must(markers.MakeDefinition("kubebuilder:selectablefield", markers.DescribesType, SelectableField{})).
 		WithHelp(SelectableField{}.Help()),
+
+	must(markers.MakeDefinition("kubebuilder:featuregate", markers.DescribesType, TypeFeatureGate(""))).
+		WithHelp(TypeFeatureGate("").Help()),
```

Comment:
Can you not register the same marker as both a Type and Field level marker without duplicating the marker as we currently have?